iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/LoadFlowBasedPhaseShifterOptimizer.java

```diff
@@ -41,16 +42,13 @@ public LoadFlowBasedPhaseShifterOptimizer(ComputationManager computationManager)
         this(computationManager, LoadFlowBasedPhaseShifterOptimizerConfig.load());
     }
 
-    private void runLoadFlow(Network network, String workingStateId) {
-        try {
-            String loadFlowName = config.getLoadFlowName().orElse(null);
-            LoadFlowResult result = LoadFlow.find(loadFlowName)
-                                            .run(network, workingStateId, computationManager, LoadFlowParameters.load());
-            if (!result.isOk()) {
-                throw new PowsyblException("Load flow diverged during phase shifter optimization");
-            }
-        } catch (Exception e) {
-            throw new PowsyblException(e);
+    private void runLoadFlow(Network network, String workingStateId, boolean throwException) {
```

Comment:
I don't think this class should be modified.
You can just add a try-catch in [`PhaseShifterOptimizeTap.apply()`](https://github.com/powsybl/powsybl-core/blob/1e0f8f569cab34a86ab13e868ea7643f3c0e2b1c/iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/PhaseShifterOptimizeTap.java#L34) around the usage of `LoadFlowBasedPhaseShifterOptimizer` and then call `logOrThrow(...)` in the catch clause.